@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -313,46 +345,118 @@ export function placePointLabel(
 
     if (label.bounds === undefined) {
         label.bounds = new THREE.Box2();
-        tmpMeasurementParams.outputCharacterBounds = undefined;
-        tmpMeasurementParams.path = undefined;
-        tmpMeasurementParams.pathOverflow = false;
-        tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
-        textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
     }
 
-    screenPosition.add(computePointTextOffset(label, scale, tmpTextOffset));
-    outScreenPosition.set(screenPosition.x, screenPosition.y, labelState.renderDistance);
-
-    // TODO: Make the margin configurable
-    tmpBox.copy(label.bounds!).expandByVector(pointLabelMargin);
-    tmpBox.min.multiplyScalar(scale);
-    tmpBox.max.multiplyScalar(scale);
-    tmpBox.translate(screenPosition);
-    tmp2DBox.set(
-        tmpBox.min.x,
-        tmpBox.min.y,
-        tmpBox.max.x - tmpBox.min.x,
-        tmpBox.max.y - tmpBox.min.y
-    );
-
-    // Check the text visibility.
-    if (!screenCollisions.isVisible(tmp2DBox)) {
-        return PlacementResult.Invisible;
-    }
+    // Setup measurements parameters for textCanvas.measureText().
+    tmpMeasurementParams.outputCharacterBounds = undefined;
+    tmpMeasurementParams.path = undefined;
+    tmpMeasurementParams.pathOverflow = false;
+    tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
```

Comment:
For persistent labels you most probably won't have to change the placement and therefore won't have to measure the text again.
This function need to be split into two: the inner one is mostly like the placePointLabel and the new one that calls placePointLabel with every placement alternative.

Reply:
I have split it in the new commit as agreed, as for persistent labels do you suggest to check run only single (current) placement version? As far as I know this will "place" a label as new in the next run (frame) searching again for the new possibilities. Is this what we are aiming for?

Reply:
Please confirm that we don't want to search for alternatives if the label is persistent. See my comments above.
The rest is done.